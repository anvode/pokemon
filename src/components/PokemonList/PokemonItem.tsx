import * as React from 'react';
import { useEffect, useState } from 'react';
import { Link } from 'react-router-dom';
import Loader from '../Loader/Loader';
import { capitalize } from '../../Helpers';

export interface PokemonItemProps {
    name: string;
    url: string
}

const PokemonItem: React.FC<PokemonItemProps> = ({ name, url }) => {
    const [imageLoading, setImageLoading] = useState(true);
    const [imageUrl, setImageUrl] = useState('');
    const index = url.split('/')[url.split('/').length - 2];

    useEffect(() => {
        setImageUrl(`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/${index}.png`);
    }, []);

    return <>
        <div key={name} className="col-md-4">
            <Link to={`/pokemon/${index}`} className="card-link d-block mb-4">
                <div className="card">
                    <div className="card-header bg-primary text-light d-flex justify-content-between">
                        <span className="pokemon__index">{index}</span>  <span className="pokemon__name">{capitalize(name)}</span>
                    </div>
                    <div className="card-img-top d-flex justify-content-center" style={{
                        height: '96px'
                    }}>
                        {imageLoading ? (
                            <Loader></Loader>
                        ) : (
                            null
                        )}
                        <img
                            className="image"
                            src={imageUrl}
                            alt={name}
                            onLoad={() => setImageLoading(false)}
                            style={
                                imageLoading
                                    ? { display: 'none' }
                                    : { display: 'block' }
                            }
                        />
                    </div>
                  
                </div>
            </Link>
        </div>
    </>;
};

export default PokemonItem;