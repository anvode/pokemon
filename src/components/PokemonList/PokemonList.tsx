import * as React from 'react';

import './PokemonList.scss';

export interface PokemonListProps {}

const PokemonList: React.FC<PokemonListProps> = () => {
    
    return <>
        <h1 className="app__title">Pokemon List</h1>
    </>;
};

export default PokemonList;