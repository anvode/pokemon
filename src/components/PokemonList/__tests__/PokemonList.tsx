import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import PokemonList from '../PokemonList';

afterEach(cleanup);

describe('PokemonList', () => {
    it('renders without crashing', () => {
        const { asFragment } = render(<PokemonList />);
        expect(asFragment()).toMatchSnapshot();
    });
});

