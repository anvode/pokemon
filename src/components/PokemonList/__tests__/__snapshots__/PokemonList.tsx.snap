// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonList renders without crashing 1`] = `
<DocumentFragment>
  <h1
    class="app__title"
  >
    Pokemon List
  </h1>
</DocumentFragment>
`;
